assets/agenda/components/AgendaPreviewEvent.tsx

```diff
@@ -24,110 +23,112 @@ class AgendaPreviewEventComponent extends React.Component<any, any> {
-                                        <i className={classNames('icon-small--arrow-down me-1', {
-                                            'rotate-90-ccw': !this.state.expanded,
-                                        })} />
-                                    </a>
-                                    <h3 onClick={this.toggleExpanded}>{getName(this.props.event)}</h3>
-                                </div>
-                                <div className="agenda-planning__preview-date">
-                                    <AgendaTime item={this.props.event}>
-                                        <AgendaListItemLabels item={this.props.event} />
-                                    </AgendaTime>
-                                </div>
-                                {!this.state.expanded ? null : (
-                                    <div className="agenda-planning__preview-metadata">
-                                        <AgendaMeta item={this.props.event} />
-                                        <AgendaLongDescription item={this.props.event} />
-                                        <AgendaPreviewAttachments item={this.props.event} />
-                                        <AgendaTags
-                                            item={this.props.event}
-                                            isItemDetail={false}
-                                        />
-                                        <AgendaEdNote
-                                            item={this.props.event}
-                                            plan={{}}
-                                            secondaryNoteField="state_reason"
-                                        />
-                                        <AgendaInternalNote
-                                            internalNote={getInternalNote(this.props.event, {})}
-                                            mt2={!!(this.props.event.ednote || this.props.event.state_reason)}
-                                        />
-                                    </div>
-                                )}
-                            </React.Fragment>
-                        )}
-                    </div>
+                    <span className="info-box__label">{gettext('Related Events')}</span>
+                    {this.state.loading ? (
+                        <div className="spinner-border text-success" />
+                    ) : (
+                        this.props.events.map((event: any) => this.renderEvent(event))
+                    )}
                 </div>
             </div>
         );
     }
 }
 
-AgendaPreviewEventComponent.propTypes = {
-    item: PropTypes.object,
-    event: PropTypes.object,
-    fetchEvent: PropTypes.func,
+const mapStateToProps = (state: any, ownProps: any) => {
+    const eventIds = ownProps.item.event_ids || [];
+    return {
+        events: eventIds.map((eventId: string) => state.itemsById[eventId]),
```

Comment:
Rather than constructing this here, which creates a new value every time, you could add `itemsById` to the `mapStateToProps` and just use it in the render function.

You will also have to check for null values, as `itemsById[eventId]` is not guarantee'd to be non-null (if the Event was not loaded for whatever reason).